### PR TITLE
add turbine dao and controller

### PR DESCRIPTION
**Pull request by adamkorynta**
updated turbine dto to support content type aliases.
added integration tests for the DAO and for the controller

### CODE REVIEW

**Comment by adamkorynta** [comment]
based on PR: https://github.com/USACE/cwms-data-api/pull/707

**Review by MikeNeilson — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/api/TurbineController.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            dao.storeTurbine(turbine, failIfExists);
+            ctx.status(HttpServletResponse.SC_CREATED).json("Created Turbine");
+        }
+
+    }
+
+    @OpenApi(
+            pathParams = {
+                    @OpenApiParam(name = NAME, description = "Specifies the name of "
+                            + "the turbine to be deleted."),
```
Comment (rma-rripken) [inline cwms-data-api/src/main/java/cwms/cda/api/TurbineController.java]: this is update not delete

Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/api/TurbineController.java]: oh  or are you saying that hte name will be deleted b/c its being replaced?  Even if that were so, rewrite that sentence.

File: cwms-data-api/src/test/java/cwms/cda/api/TurbineControllerIT.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    }
+
+    @Test
+    void test_update_does_not_exist() {
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SWT_NORMAL;
+        //Create the Turbine
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(Formats.JSON)
+            .contentType(Formats.JSON)
```
Comment (rma-rripken) [inline cwms-data-api/src/test/java/cwms/cda/api/TurbineControllerIT.java]: contentType probably not needed here.

File: cwms-data-api/src/main/java/cwms/cda/api/TurbineController.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/api/TurbineController.java]: naw, just a bad copy/paste. Updated

File: cwms-data-api/src/test/java/cwms/cda/api/TurbineControllerIT.java (hunk 2 above)
Comment (adamkorynta) [reply cwms-data-api/src/test/java/cwms/cda/api/TurbineControllerIT.java]: cleaned up the accept vs content type in this and the embankments

**Review by rma-rripken — DISMISSED** [review]

**Review by rma-rripken — DISMISSED** [review]

**Review by rma-rripken — APPROVED** [review]

**Pull request merged**